spec/emerald/parser_spec.rb

```diff
@@ -164,6 +164,23 @@
                                            Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(3),
                                                             Emerald::List.new(  Emerald::Atom.new('-'), Emerald::Number.new(4) ) ) ) ] )
     end
+    
+    it "parses a list which contains a list and a number to the right of the internal list" do
+      source = '(+ (- 0 1) 2))'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::List.new( Emerald::Atom.new('+'),  
+                                           Emerald::List.new( Emerald::Atom.new('-'), Emerald::Number.new(0), Emerald::Number.new(1) ),
+                                                             Emerald::Number.new(2) ) ] )
+    end
+
+    it "parses nested lists with numbers to the left of the lists" do
+      source = '(+ 2 (+ 3 (+ 0) 4) 3 4)'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(2), 
+                                           Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(3) , 
+                                                             Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(0) ),
+                                                             Emerald::Number.new(4) ), Emerald::Number.new(3), Emerald::Number.new(4) ) ] )
```

Comment:
These lines are getting really long. Ideally we'd keep it within 80 or 100 characters (depending on the project code style). Worth adding a linter like Rubocop?